### DISCUSSION TITLE
Improve lib keycloak-angular to be installed in Angular 18

### DISCUSSION THREAD

**Discussion by tiagonic** [keycloak/keycloak#29845] [Ideas]
In front-end application with Angular 18.0.1

_Angular CLI: 18.0.1
Node: 20.12.2
Package Manager: npm 10.7.0
OS: win32 x64

Angular: 18.0.0
... animations, cdk, common, compiler, compiler-cli, core, forms
... material, platform-browser, platform-browser-dynamic
... platform-server, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1800.1
@angular-devkit/build-angular   18.0.1
@angular-devkit/core            18.0.1
@angular-devkit/schematics      18.0.1
@angular/cli                    18.0.1
@angular/ssr                    18.0.1
@schematics/angular             18.0.1
rxjs                            7.8.1
typescript                      5.4.5
zone.js                         0.14.6_

When execute `npm install keycloak-angular` the follow error is retorned:

"npm install keycloak-angular
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: app@0.0.0
npm error Found: @angular/common@18.0.0
npm error node_modules/@angular/common
npm error   @angular/common@"^18.0.0" from the root project
npm error
npm error Could not resolve dependency:
npm error peer @angular/common@"^17" from keycloak-angular@15.2.1
npm error node_modules/keycloak-angular
npm error   keycloak-angular@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error C:\Users\Tiago\AppData\Local\npm-cache\_logs\2024-05-24T16_14_46_059Z-eresolve-report.txt

npm error A complete log of this run can be found in: C:\Users\Tiago\AppData\Local\npm-cache\_logs\2024-05-24T16_14_46_059Z-debug-0.log"

How to fix this?

**Comment by jonkoops** [comment]
The Keycloak project is not affiliated with the Keycloak Angular project, so this is not the place to ask for support. That said, if you read the logs:

> npm error Could not resolve dependency:
> npm error peer @angular/common@"^17" from keycloak-angular@15.2.1

You'll see that `keycloak-angular` wants `@angular/common` version 17.x, but you have provided 18.x, which Keycloak Angular [does not yet support](https://github.com/mauriciovigolo/keycloak-angular?tab=readme-ov-file#versions).
